### DISCUSSION TITLE
Site fails to buid with error: `error calling len: reflect: call of reflect.Value.Type on zero Value` for contributors value in blog template.

### DISCUSSION THREAD

**Discussion by anoduck** [thuliteio/doks#1279] [Q&A] [answered]
Migrating site to doks and managed to get three pages into the migration, then hugo dev server crashed with
the below error. I am unsure how to troubleshoot the problem, because doks is packaged via npm and therefore
does not have the theme template files directly accessible. 

I admit to attempting to setup the vscode extension "Frontmatter", to act as a embeded CMS for the site, but I
have reviewed and cleaned up the front matter generated by the extension, and it passes the mustard.

The output does indicate the error is associated with recieving a zero length value for "contributors" in the
blog template.  I am assuming the front matter of doks is in yaml format, not toml, and all "blog" pages
possess a defined "contributor". 

```bash
WARN  deprecated: .Site.IsMultiLingual was deprecated in Hugo v0.124.0 and will be removed in a future release. Use hugo.IsMultilingual instead.
ERROR render of "term" failed: "/website/node_modules/@thulite/doks-core/layouts/_default/term.html:47:15": 
    execute of template failed: template: _default/term.html:47:15:
        executing "main" at <partial "main/blog-meta.html" .>: 
            error calling partial: "/website/node_modules/@thulite/doks-core/layouts/partials/main/blog-meta.html:1:17": 
                execute of template failed: template: partials/main/blog-meta.html:1:17: 
                    executing "partials/main/blog-meta.html" at <len .Params.contributors>:
                        error calling len: reflect: call of reflect.Value.Type on zero Value
INFO  build:  step render pages 37 content 33 duration 4.315598387s
INFO  build:  step render deferred count 0 duration 4.126Âµs
INFO  build:  step postProcess duration 19.148945ms
INFO  build:  duration 4.550655948s
Built in 5303 ms
Error: error building site: render: failed to render pages: 
    render of "term" failed: "/website/node_modules/@thulite/doks-core/layouts/_default/term.html:47:15": 
        execute of template failed: template: _default/term.html:47:15: executing "main" at <partial "main/blog-meta.html" .>: 
            error calling partial: "/website/node_modules/@thulite/doks-core/layouts/partials/main/blog-meta.html:1:17":
                execute of template failed: template: partials/main/blog-meta.html:1:17: executing "partials/main/blog-meta.html" at <len .Params.contributors>: 
                    error calling len: reflect: call of reflect.Value.Type on zero Value
```

**Comment by h-enk** [comment]
Add `contributors: []` (which does not need to be defined but needs to be there) to the front matter of your blog pages. For example (if you're using `yaml`):

```yaml
contributors: []
```

**Reply by h-enk** [reply]
It should work like you expected — https://github.com/thuliteio/doks-core/issues/107

**Reply by anoduck** [reply]
The error message is still appearing. 

I am going to backtrack my steps, like I mentioned, I am only three posts in and have them saved elsewhere. I will remove each post individually and see at which point the build process is successful. If the build is not successful after this point, I will backup the configuration directory and do the same from there. After this, if the build is still not successful, I will scrap the project directory and start over a fresh, and avoid using the "Frontmatter" extension from here on out.

Thanks for the fast response.

**Reply by anoduck** [reply, marked as answer]
That didn't take nearly as much time as I thought, and I don't know why I didn't think to do that last night. I found the buggers, it was the "terms of service" page and the "about us" page. 

Needed to define the type. Apparently, not doing so resulted in the pages being interpreted as blog posts. I guess...

**Another important notation:** Doks uses a special type just for pages with legal content, `type: "legal"`, such as "Privacy Policy" and "Terms of Service". Application of this type causes the page to be center aligned, and applies various other style settings.

```yaml
type: "legal"
```

**Reply by anoduck** [reply]
I think I am getting "type" and "kind" confused with each other.

Regardless, the addition worked.
